src/mon/Monitor.cc

```diff
@@ -5337,6 +5337,16 @@ void Monitor::count_metadata(const string& field, Formatter *f)
   f->close_section();
 }
 
+void Monitor::check_daemon_versions(map<string,map<int, int>, std::greater<std::string>> &versions){
```

Comment:
Why isn't `versions` a `map<string, std::list<string> >`. Where the _first_ string is the version and _second_ is a list of daemons?   Maybe the map sort by string order is sufficient to be in version order?

Also, why isn't hostname needed here to get the p.first converted to osd.0, mon.a...?  This was mentioned in a comment see `Monitor::print_nodes()` in `mon/Monitor.cc`.

Reply:
By using greater in the below code it automatically sorts the versions which are stored as a string in order from largest to lowest:
std::map<string, map<int, int>, std::greater<std::string>> multiple_versions;